spring-cloud-consul-core/src/test/java/org/springframework/cloud/consul/ConsulHealthIndicatorTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.Response;
+import com.ecwid.consul.v1.catalog.CatalogServicesRequest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link ConsulHealthIndicator}.
+ *
+ * @author Chris Bono
+ */
+@RunWith(SpringRunner.class)
+public class ConsulHealthIndicatorTest {
```

Comment:
I wanted to test the variants that were added in this code proposal. I first looked at doing in the integration tests w// similar mocks (`@MockBean`) but it became clear they did not belong there. I attempted to balance the level of detail to just mock and test the bare minimum we need to avoid a super brittle test that is in lockstep w/ the indicator impl. I think its not too bad and worth the investment.

Reply:
I think I can simplify this by using the existing integration up/down test and adding `@SpyBean/@MockBean`. I will see how it looks in code and then may update when I make the other suggested changes.